### PR TITLE
SW-3932 Editable boundaries for simple planting sites

### PR DESCRIPTION
**Pull request by sgrimm**
Add a "Simple Map Editor" opt-in feature.

If enabled, when editing or creating a simple planting site, the UI will include
an editable map where the user can add a polygon or edit an existing one to define
the planting site's boundary.

### CODE REVIEW

**Comment by sgrimm** [comment]
Rebased this on main to pick up #1671 and #1673

File: src/components/Map/EditableMapDraw.tsx [hunk 1]
```diff
@@ -0,0 +1,229 @@
...
+        },
+        positionOptions: {
+          enableHighAccuracy: true,
+        },
+      })
+  );
+
+  useControl<ScaleControl>(() => new ScaleControl());
+
+  return null;
```
Comment (charlie-tf) [inline src/components/Map/EditableMapDraw.tsx:228]: It's interesting to define a component and use it for only its hooks, returning `null` at the end. Looks fine to me though—it well encapsulates all the controls logic, separating it from `EditableMap` 👍

**Review by charlie-tf — APPROVED** [review]
Couple minor comments. Looks great, thanks!

File: src/components/Map/EditableMap.tsx [hunk 2]
```diff
@@ -0,0 +1,116 @@
...
+export type EditableMapProps = {
+  boundary?: MultiPolygon;
+  onBoundaryChanged: (boundary: MultiPolygon | null) => void;
+  style?: object;
+};
+
+const polygonIcon = <button className='mapbox-gl-draw_polygon mapbox-gl-draw_ctrl-draw-btn' />;
+const trashIcon = <button className='mapbox-gl-draw_trash mapbox-gl-draw_ctrl-draw-btn' />;
+
+function instructionsForMode(mode: MapEditorMode | null): JSX.Element | JSX.Element[] | null {
```
[the remark this answers is not in the document]
Comment (sgrimm) [reply src/components/Map/EditableMap.tsx:20]: Good catch. This was left over from earlier versions of the function.

**Review by charlie-tf — APPROVED** [review]

**Pull request merged**